### PR TITLE
[Mellanox] Change SAI build procedure

### PR DESCRIPTION
**Pull request by keboliu**
Change SAI build procedure.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: platform/mellanox/mlnx-sai.mk [hunk 1]
```diff
@@ -11,4 +24,16 @@ $(MLNX_SAI)_RDEPENDS += $(MLNX_SDK_RDEBS) $(MLNX_SDK_DEBS)
...
+
+endef
+
+$(eval $(foreach deb,$(MLNX_SAI) $(MLNX_SAI_DBGSYM),$(call make_url,$(deb))))
+
+ifeq ($(SAI_FROM_SRC), y)
 SONIC_MAKE_DEBS += $(MLNX_SAI)
+else
+SONIC_ONLINE_DEBS += $(MLNX_SAI)
+endif
```
Comment (Yakiv-Huryk) [inline platform/mellanox/mlnx-sai.mk]: please add an empty line at the end of a file

File: platform/mellanox/mlnx-sai/Makefile [hunk 2]
```diff
@@ -7,6 +7,8 @@ DERIVED_TARGETS = mlnx-sai-dbgsym_1.mlnx.$(MLNX_SAI_VERSION)_$(CONFIGURED_ARCH).
 
 $(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
 	pushd SAI-Implementation
```
Comment (Yakiv-Huryk) [inline platform/mellanox/mlnx-sai/Makefile]: Do we need this directory? It's empty now, and the only use it is to store the downloaded mlnx_sai temporarily.

File: platform/mellanox/mlnx-sai.mk [hunk 3]
```diff
@@ -1,8 +1,21 @@
 # Mellanox SAI
 
 MLNX_SAI_VERSION = SAIBuild2305.24.0.1
+MLNX_SAI_ASSETS_GITHUB_URL = https://github.com/Mellanox/Spectrum-SDK-Drivers-SONiC-Bins
+MLNX_SAI_ASSETS_RELEASE_TAG = sai-$(MLNX_SAI_VERSION)-$(BLDENV)-$(CONFIGURED_ARCH)
+MLNX_SAI_ASSETS_URL = $(MLNX_ASSETS_GITHUB_URL)/releases/download/$(MLNX_SAI_ASSETS_RELEASE_TAG)
+MLNX_SAI_DEB_VERSION = $(subst -,.,$(subst _,.,$(MLNX_SAI_VERSION)))
 
-export MLNX_SAI_VERSION
+# Place here URL where SDK sources exist
```
Comment (Yakiv-Huryk) [inline platform/mellanox/mlnx-sai.mk]: *SAI* sources probably

**Comment by keboliu** [comment]
@Yakiv-Huryk review comments addressed, would you please help to check?

**Review by Yakiv-Huryk — APPROVED** [review]

**Pull request closed without merging**